src/modules/iotjs_module_mqtt.c

```diff
@@ -0,0 +1,777 @@
+      sizeof(packet_identifier_msb) + sizeof(packet_identifier_lsb);
+  uint32_t remaining_length = payload_len + variable_header_len;
+  size_t full_len = sizeof(header_byte) + payload_len + variable_header_len +
+                    get_remaining_length_size(remaining_length);
+
+  jerry_value_t jbuff = iotjs_bufferwrap_create_buffer(full_len);
+  iotjs_bufferwrap_t *buffer_wrap = iotjs_bufferwrap_from_jbuffer(jbuff);
+
+  uint8_t *buff_ptr = (uint8_t *)buffer_wrap->buffer;
+
+  *buff_ptr++ = header_byte;
+
+  buff_ptr += iotjs_encode_remaining_length(remaining_length, buff_ptr);
+
+  *buff_ptr++ = packet_identifier_msb;
+  *buff_ptr++ = packet_identifier_lsb;
+
+  memcpy(buff_ptr, topic_payload.buffer, topic_payload.buffer_size);
+  buff_ptr += topic_payload.buffer_size;
+
+  return jbuff;
+}
+
+
+JS_FUNCTION(MqttDisconnect) {
+  DJS_CHECK_THIS();
+
+  uint8_t header_byte = 0;
+  uint8_t remaining_length = 0;
+  header_byte |= (DISCONNECT << 4);
+
+  size_t full_len = sizeof(header_byte) + sizeof(remaining_length);
+
+  jerry_value_t jbuff = iotjs_bufferwrap_create_buffer(full_len);
+  iotjs_bufferwrap_t *buffer_wrap = iotjs_bufferwrap_from_jbuffer(jbuff);
+
+  uint8_t *buff_ptr = (uint8_t *)buffer_wrap->buffer;
+
+  *buff_ptr++ = header_byte;
+  *buff_ptr++ = remaining_length;
+
+  return jbuff;
+}
+
+JS_FUNCTION(MqttSendAck) {
+  DJS_CHECK_THIS();
+  DJS_CHECK_ARGS(1, object);
+
+  jerry_value_t joptions = JS_GET_ARG(0, object);
+  jerry_value_t jack_type = iotjs_jval_get_property(joptions, IOTJS_MAGIC_STRING_ACKTYPE);
+  jerry_value_t jpacket_id = iotjs_jval_get_property(joptions, IOTJS_MAGIC_STRING_PACKETID);
+
+  uint16_t packet_id = 0;
+  if (!jerry_value_is_undefined(jpacket_id)) {
+      packet_id = (uint16_t) jerry_get_number_value(jpacket_id);
+  }
+
+  uint8_t ack_type = 0;
+  if (!jerry_value_is_undefined(jack_type)) {
+      ack_type = (uint8_t) jerry_get_number_value(jack_type);
+  }
+
+  uint8_t header_byte = 0;
+  uint8_t remaining_length = 2;
+  
+  switch(ack_type) {
+    case PUBACK: {
+      header_byte |= (PUBACK << 4);
+      break;
+    }
+    case PUBREC: {
+      header_byte |= (PUBREC << 4);
+      break;
+    }
+    case PUBREL: {
+      header_byte |= (PUBREL << 4);
+      header_byte |= 2;
+      break;
+    }
+    case PUBCOMP: {
+      header_byte |= (PUBCOMP << 4);
+      break;
+    }
+  }
+
+  uint8_t packet_identifier_lsb = 0;
+  uint8_t packet_identifier_msb = 0;
+  iotjs_mqtt_calculate_msb_lsb(
+    packet_id, &packet_identifier_msb, &packet_identifier_lsb);
+
+  size_t full_len = sizeof(header_byte) + sizeof(remaining_length)
+   + sizeof(packet_id);
+
+  jerry_value_t jbuff = iotjs_bufferwrap_create_buffer(full_len);
+  iotjs_bufferwrap_t *buffer_wrap = iotjs_bufferwrap_from_jbuffer(jbuff);
+
+  uint8_t *buff_ptr = (uint8_t *)buffer_wrap->buffer;
+
+  *buff_ptr++ = header_byte;
+  *buff_ptr++ = remaining_length;
+  *buff_ptr++ = packet_identifier_msb;
+  *buff_ptr++ = packet_identifier_lsb;
```

Comment:
For the last bytes you can use `buff_ptr[0]` - `buff_ptr[3]`. The last ++ has no effect anyway.